### PR TITLE
Ignore keyboard and mouse if egui uses them

### PR DESCRIPTION
**Pull request by Shatur**

### CODE REVIEW

File: src/systems.rs [hunk 1]
```diff
@@ -76,19 +79,28 @@ pub fn update_action_state<A: Actionlike>(
...
-    let mouse_wheel = mouse_wheel.into_inner();
+    let keycodes = keycodes.map(|keycodes| keycodes.into_inner());
+    let mouse_buttons = mouse_buttons.map(|mouse_buttons| mouse_buttons.into_inner());
+    let mouse_wheel = mouse_wheel.map(|mouse_wheel| mouse_wheel.into_inner());
     let mouse_motion = mouse_motion.into_inner();
 
+    #[cfg(feature = "egui")]
+    let (keycodes, mouse_buttons, mouse_wheel) = {
+        let ctx = egui.ctx_mut();
+        let keycodes = keycodes.filter(|_| !ctx.wants_keyboard_input());
```
Comment (alice-i-cecile) [inline src/systems.rs:90]:
```suggestion
        // If egui wants to own inputs, don't also apply them to the game state
        let keycodes = keycodes.filter(|_| !ctx.wants_keyboard_input());
```

File: Cargo.toml [hunk 2]
```diff
@@ -20,11 +20,13 @@ members = ["./", "tools/ci", "macros"]
 [features]
 default = ['ui']
 ui = ['bevy/bevy_ui']
+egui = ['dep:bevy_egui']
```
Comment (alice-i-cecile) [inline Cargo.toml:23]:
```suggestion
# If this feature is enabled, egui will have priority over actions when processing inputs
egui = ['dep:bevy_egui']
```

**Review by alice-i-cecile — CHANGES_REQUESTED** [review]
A couple of comments, and needs a release note section :)

File: Cargo.toml (hunk 2 above)
Comment (Shatur) [reply Cargo.toml:23]: Agree, worth having!

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**